### ISSUE TITLE
Applying online softmax patterns on joint_graph cause 1.2x peak memory regression for TB hf_T5_base model

### ISSUE THREAD

**Issue by shunting314** [pytorch/pytorch#148141]
### 🐛 Describe the bug

Command 
```time python benchmarks/dynamo/torchbench.py --backend inductor --amp --performance --only hf_T5_base --training```

tlparse for joint-graph pattern: https://manifold.edge.x2p.facebook.net/v0/read/tree/logs/.tmpk7zTmz/index.html?bucketName=tlparse_reports&apiKey=tlparse_reports-key&withPayload=1&timeoutMsec=100 . Peak memory 22.2GB.

tlparse for post-grad graph pattern: https://manifold.edge.x2p.facebook.net/v0/read/tree/logs/.tmp9eCEvc/index.html?bucketName=tlparse_reports&apiKey=tlparse_reports-key&withPayload=1&timeoutMsec=100  . Peak memory 18.6GB.

It seems like the partitioner makes worse decisions if the pattern is applied before rather than after the partitioning.

This is related to PR https://github.com/pytorch/pytorch/pull/127011 . cc @chauhang @penguinwu @jansel @eellison I'll move the pattern to post grad to avoid the regression. Ultimately we probably should improve the partitioner. cc @Chillee as well

### Versions

.

cc @chauhang @penguinwu
